src/components/Carousel.tsx

```diff
@@ -539,6 +547,43 @@ export default class Carousel extends React.Component<Props, State> {
         return hasMoved;
     };
 
+    getState = () => {
+        const stateObject: Readonly<State> = { ...this.state };
```

Comment:
From what I can see, this method is only called to retrieve the `selectedItem`. It's traversing over all the state attributes on every call, just for that. It's kind of a waste, don't you think?

Reply:
Either we create a getter just for the `selectedItem` or an array with all controlled props (only `selectedItem` right now) to iterate over instead of iterating over the whole state.... thoughts?

Reply:
I implemented this way to make the `getState` method be used just like `this.state`, but it's really a waste, for now, to use it to get only `selectedItem`. I think that we can use the second solution on `getState` and merge it with the state object, it would keep the `getState` purpose.

We may also want to change all `this.state` occurrences to `getState` since it would not be wasteful as before, then we would never need to think which one to use.